src/service/viewport-impl.js

```diff
@@ -1065,6 +1070,156 @@ export class ViewportBindingNaturalIosEmbed_ {
 }
 
 
+export class ViewportBindingNaturalIosScrollEmbed_
+    extends ViewportBindingNaturalIosEmbed_ {
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    super(win);
+
+    // TODO
+    this.onScroll();
+
+    this.onResize(() => this.onResized_());
+  }
+
+  /** @override */
+  requiresFixedLayerTransfer() {
+    return false;
+  }
+
+  /** @override */
+  requiresFixedLayerLookup() {
+    return true;
+  }
+
+  /** @private */
+  setup_() {
+    const doc = this.win.document;
+    const documentElement = doc.documentElement;
+    const body = doc.body;
+
+    // Setup scrolling styles.
+    // html {
+    //   height: 100% !important;
+    //   -webkit-overflow-scrolling: touch;
+    // }
+    // body {
+    //   height: 100% !important;
+    //   overflow-x: hidden;
+    // }
```

Comment:
Let's include here also `i-amp-container` styles and why it's needed in some detail.